flink-connectors/flink-connector-files/src/test/java/org/apache/flink/architecture/TestCodeArchitectureTest.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.architecture;
+
+import org.apache.flink.architecture.common.ImportOptions;
+import org.apache.flink.architecture.rules.ITCaseRules;
+
+import com.tngtech.archunit.core.importer.ImportOption;
+import com.tngtech.archunit.junit.AnalyzeClasses;
+import com.tngtech.archunit.junit.ArchTest;
+import com.tngtech.archunit.junit.ArchTests;
+
+/** Architecture tests for test code. */
+@AnalyzeClasses(
+        packages = "org.apache.flink.connector.file",
+        importOptions = {
+            ImportOption.OnlyIncludeTests.class,
+            ImportOptions.ExcludeScalaImportOption.class,
+            ImportOptions.ExcludeShadedImportOption.class
+        })
+public class TestCodeArchitectureTest {
+
+    @ArchTest public static final ArchTests ITCASE = ArchTests.in(ITCaseRules.class);
```

Comment:
hmm, generally, composition should be always over inheritance. I would not create a new inheritance, especially only to reduce the imports.

Reply:
I agree on composition over inheritance, but I want to also stay DRY; if we introduce rules for, say, unit tests, every module has to import these rules individually now.

How about instead of a base class to inherit from, we make a `StandardTestCodeTests` class that imports `ItCaseRules`, and then each module imports that class via `@ArchTest`? Then we can extend `StandardTestCodeTests` with rule sets for unit tests (or other things).

Reply:
fair enough